internal/lang/funcs/filesystem.go

```diff
@@ -58,6 +61,50 @@ func MakeFileFunc(baseDir string, encBase64 bool) function.Function {
 	})
 }
 
+func templateMaxRecursionDepth() (int, error) {
+	envkey := "TF_TEMPLATE_RECURSION_DEPTH"
```

Comment:
I wonder if this shouldn't be hard coded. It seems like a footgun to change this because it would make modules break under certain situations.

Reply:
The default limit (1024) should work for all of the proposed scenarios so far.  However, I usually like to provide escape hatches for users where possible (even if they are not recommended).  Would documentation around using this environment variable as a last resort assuage your concerns?